app/scripts/lib/multichain-api/caip25permissions.ts

```diff
@@ -94,12 +95,27 @@ const specificationBuilder: PermissionSpecificationBuilder<
         throw new Error('missing expected caveat values'); // TODO: throw better error here
       }
 
-      const processedScopes = processScopes(requiredScopes, optionalScopes, {
-        findNetworkClientIdByChainId,
+      const { flattenedRequiredScopes, flattenedOptionalScopes } =
+        processScopes(requiredScopes, optionalScopes);
+
+      const existsNetworkClientForChainId = (chainId: Hex) => {
+        try {
+          findNetworkClientIdByChainId(chainId);
+          return true;
+        } catch (err) {
+          return false;
+        }
+      };
+
+      assertScopesSupported(flattenedRequiredScopes, {
+        existsNetworkClientForChainId,
+      });
+      assertScopesSupported(flattenedOptionalScopes, {
+        existsNetworkClientForChainId,
       });
 
-      assert.deepEqual(requiredScopes, processedScopes.flattenedRequiredScopes);
-      assert.deepEqual(optionalScopes, processedScopes.flattenedOptionalScopes);
+      assert.deepEqual(requiredScopes, flattenedRequiredScopes);
+      assert.deepEqual(optionalScopes, flattenedOptionalScopes);
```

Comment:
do we need to make this assertion for optionalScopes?

Reply:
Yes. I don't see why we wouldn't

Reply:
actually wait stepping back what is this assertion achieving?

Reply:
that the required and optional scopes being persisted have had all validations and transforms (flattening and merging) applied to them. `processScopes` is idempotent. Running the output of `processScopes` again through `processScopes` should return the same result. The deepEqual asserts check that

Reply:
For some reason I'm just not seeing where the first args here `requiredScopes` and `optionalScopes` are processed themselves?

Reply:
Oh is the idea that this validator method is verifying that they've been processed in the same manner. Ah ok and then `processScopes` is idempotent so it shouldn't change them. Ok I think I understand